Mathlib/Analysis/SpecialFunctions/MulExpNegMulSqIntegral.lean

```diff
@@ -0,0 +1,247 @@
+  respect to a finite measure `P` can be approximated by the integral of the sequence approximating
+  the exponential function, `fun x => (g * (1 + (n : ℝ)⁻¹ • -(ε • g * g)) ^ n) x`. This allows to
+  transfer properties of a subalgebra of functions containing `g` to the function
+  `mulExpNegMulSq ε ∘ g`, see e.g. `integral_mulExpNegMulSq_comp_eq`.
+
+## Main Result
+
+`dist_integral_mulExpNegMulSq_comp_le`: For a subalgebra of functions `A`, if for any `g ∈ A` the
+integral with respect to two finite measures `P, P'` coincide, then the difference of the integrals
+of `mulExpNegMulSq ε ∘ g` with respect to `P, P'` is bounded by `6 * sqrt ε`.
+This will be a key ingredient in the proof of theorem `ext_of_forall_mem_subalgebra_integral_eq`
+(future work), where it is shown that a subalgebra of functions that separates points separates
+finite measures.
+-/
+
+open MeasureTheory Real NNReal ENNReal BoundedContinuousFunction Filter
+
+open scoped Topology
+
+variable {E : Type*} [TopologicalSpace E] [MeasurableSpace E] [BorelSpace E]
+    {P : Measure E} [IsFiniteMeasure P] {ε : ℝ}
+
+theorem integrable_mulExpNegMulSq_comp (f : C(E, ℝ)) (hε : 0 < ε) :
+    Integrable (fun x => mulExpNegMulSq ε (f x)) P := by
+  apply integrable P ⟨⟨fun x => mulExpNegMulSq ε (f x), by fun_prop⟩, ⟨2 * (sqrt ε)⁻¹, _⟩⟩
+  exact fun x y => dist_mulExpNegMulSq_le_two_mul_sqrt hε (f x) (f y)
+
+theorem integrable_mulExpNegMulSq_comp_restrict_of_isCompact {K : Set E} (hK : IsCompact K)
+    (hKmeas : MeasurableSet K) (g : C(E, ℝ)) :
+    Integrable (fun x => mulExpNegMulSq ε (g x)) (P.restrict K) :=
+  g.continuous.mulExpNegMulSq.continuousOn.integrableOn_compact' hK hKmeas
+
+/-- The integral of `mulExpNegMulSq ε ∘ g` with respect to a finite measure `P` converges to the
+integral of `g`, as `ε → 0` from above. -/
+theorem tendsto_integral_mulExpNegMulSq_comp (g : E →ᵇ ℝ) :
+    Tendsto (fun ε => ∫ x, mulExpNegMulSq ε (g x) ∂P) (𝓝[>] 0) (𝓝 (∫ x, g x ∂P)) := by
+  apply tendsto_of_seq_tendsto
+  intro u hu
+  obtain ⟨N, hupos⟩ := eventually_atTop.mp (tendsto_nhdsWithin_iff.mp hu).2
+  apply tendsto_integral_filter_of_norm_le_const ?h_meas ?h_bound ?h_lim
+  · exact Eventually.of_forall (fun n => g.continuous.mulExpNegMulSq.aestronglyMeasurable)
+  · use norm g
+    rw [eventually_atTop]
+    use N
+    intro n hn
+    exact Eventually.of_forall
+      (fun _ => abs_mulExpNegMulSq_comp_le_norm g (le_of_lt (Set.mem_Ioi.mp (hupos n hn))))
+  · exact Eventually.of_forall (fun _ => (tendsto_nhdsWithin_of_tendsto_nhds
+        tendsto_mulExpNegMulSq).comp hu)
+
+/-- The integral of `mulExpNegMulSq ε ∘ g` with respect to a finite measure `P` can be
+approximated by the integral of the sequence approximating the exponential function. -/
+theorem tendsto_integral_mul_one_plus_inv_mul_sq (g : E →ᵇ ℝ) (hε : 0 < ε) :
+    Tendsto (fun (n : ℕ) => ∫ x, (g * (1 + (n : ℝ)⁻¹ • -(ε • g * g)) ^ n) x ∂ P)
```

Comment:
There is no `mul` in the second term of the integral, but a `smul`. Also I would add a `pow` in there.
```suggestion
theorem tendsto_integral_mul_one_plus_inv_smul_pow_sq (g : E →ᵇ ℝ) (hε : 0 < ε) :
    Tendsto (fun (n : ℕ) => ∫ x, (g * (1 + (n : ℝ)⁻¹ • -(ε • g * g)) ^ n) x ∂P)
```